controllers/resource_controller.go

```diff
@@ -243,7 +243,7 @@ func (r *ResourceReconciler) reconcile(ctx context.Context, obj *v1alpha1.Resour
 		identity[k] = v
 	}
 
-	snapshotName, err := snapshot.GetSnapshotNameForObject(obj.GetName(), obj)
+	snapshotName, err := snapshot.GetSnapshotNameForObject(obj.GetName(), snapshot.WithSnapshotName(obj.GetSnapshotName()))
```

Comment:
This is really confusing. We are passing in an option that overrides the snapshot name. Why call the function at all if this option is set? I'm not sure I follow the intention here.

Reply:
You're right. I just thought you might like to keep the ability of defining it If it is set without having to use an if elsewhere. But you're right, this could be an if in here and make the thing even more simplified. :D

Reply:
yeah i think so 😄

Reply:
so the others didn't make any sense though because you already had a `== ""` in front of it